client/events/sockets.mjs

```diff
@@ -216,6 +218,30 @@ export default (game) => {
     game.mainPlayer.actionEnd();
   });
 
+  game.socket.on("action:rejected", (message) => {
+    const player = game.mainPlayer;
+    TextTween({
+      scene: game,
+      x: player.x,
+      y: player.y,
+      depth: player.depth,
+      message: MESSAGES[message],
+      duration: 3000,
+    });
+  });
+
+  game.socket.on("crafting:rejected", (message) => {
+    const player = game.mainPlayer;
+    TextTween({
+      scene: game,
+      x: player.x,
+      y: player.y,
+      depth: player.depth,
+      message: MESSAGES[message],
+      duration: 3000,
+    });
+  });
```

Comment:
Why not one event for this type of messages from server?
message:sent or something like that

Reply:
This way another logic on client side may be implemented to handle two separate scenarios, let me know if you still want to merge it.